pandas/core/arrays/datetimelike.py

```diff
@@ -620,9 +620,6 @@ def _values_for_factorize(self):
     def _from_factorized(cls, values, original):
         return cls(values, dtype=original.dtype)
 
-    def _values_for_argsort(self):
```

Comment:
This seems to cast to object dtype for datetimetz. I think this can be reverted.

```python
In [38]: a = pd.date_range('2000', periods=100, tz='US/Eastern')._data

In [39]: %timeit a.argsort()
325 µs ± 11.3 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

In [40]: b = pd.date_range('2000', periods=100)._data

In [41]: %timeit b.argsort()
44.8 µs ± 1.76 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```

This should be reve

Reply:
@makbigc I believe you reverted this and pref is back to normal, can you confirm (also do we hav an asv for this example that @TomAugspurger gave; if not can you add).